File: backend/app/schemas/job.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime

class ExternalJob(BaseModel):
    external_id: str
    title: str
    company: str
    description: str
    location: Optional[str] = None
    salary_min: Optional[float] = None
    salary_max: Optional[float] = None
    job_type: Optional[str] = None
    requirements: Optional[str] = None
    skills: Optional[str] = None
    source: Optional[str] = None
    apply_url: str
    deadline: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/Job.ts
export interface Job {
  external_id: string;
  title: string;
  company: string;
  description: string;
  requirements?: string;
  skills?: string;
  location?: string;
  salary_min?: number;
  salary_max?: number;
  job_type?: string;
  source?: string;
  apply_url: string;
  deadline?: string;
}

